internal/controller/spinapp_controller.go

```diff
@@ -383,12 +393,7 @@ func constructDeployment(ctx context.Context, app *spinv1alpha1.SpinApp, config
 	generatedRuntimeConfigSecretName, caSecretName string, scheme *runtime.Scheme) (*appsv1.Deployment, error) {
 	// TODO: Once we land admission webhooks write some validation to make
 	// replicas and enableAutoscaling mutually exclusive.
```

Comment:
This comment is irrelevant now. We can also probably just set this value in the deployment directly now and don't need this intermediary variable.